Frameworks/UIKit/NSString+UIKitAdditions.mm

```diff
@@ -161,48 +88,78 @@ - (void)drawInRect:(CGRect)rect withAttributes:(NSDictionary*)attrs {
 /**
  @Status Interoperable
 */
-- (CGSize)drawInRect:(CGRect)rct withFont:(UIFont*)font lineBreakMode:(UILineBreakMode)lineBreakMode alignment:(UITextAlignment)alignment {
-    CGSize fontExtent;
-
-    drawString(font, UIGraphicsGetCurrentContext(), self, rct, lineBreakMode, alignment, &fontExtent);
-
-    return fontExtent;
+- (CGSize)drawInRect:(CGRect)rect withFont:(UIFont*)font {
+    return [self drawInRect:rect withFont:font lineBreakMode:UILineBreakModeWordWrap];
 }
 
 /**
  @Status Interoperable
 */
-- (CGSize)drawInRect:(CGRect)rct withFont:(UIFont*)font lineBreakMode:(UILineBreakMode)lineBreakMode {
-    CGSize fontExtent;
-
-    drawString(font, UIGraphicsGetCurrentContext(), self, rct, lineBreakMode, UITextAlignmentLeft, &fontExtent);
-
-    return fontExtent;
+- (CGSize)drawInRect:(CGRect)rect withFont:(UIFont*)font lineBreakMode:(UILineBreakMode)lineBreakMode {
+    return [self drawInRect:rect withFont:font lineBreakMode:lineBreakMode alignment:UITextAlignmentLeft];
 }
 
 /**
  @Status Interoperable
 */
-- (CGSize)drawAtPoint:(CGPoint)pt withFont:(UIFont*)font {
-    CGSize fontExtent;
+- (CGSize)drawInRect:(CGRect)rect withFont:(UIFont*)font lineBreakMode:(UILineBreakMode)lineBreakMode alignment:(UITextAlignment)alignment {
+    CGContextRef context = UIGraphicsGetCurrentContext();
+    if (rect.size.width == 0) {
+        rect.size.width = std::numeric_limits<CGFloat>::max();
+    }
```

Comment:
is the intended behavior is to extend the region, specifically if one of the constraints are 0?